bindings/muen/muen-platform_lifecycle.c

```diff
@@ -18,18 +18,37 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include "../hvt/bindings.h"
+#include "bindings.h"
+
+struct mft *muen_manifest = NULL;
 
 void fpu_init(void)
 {
     const unsigned default_mxcsr = 0x1f80;
     __asm__ __volatile__("ldmxcsr %0" : : "m"(default_mxcsr));
 }
 
+extern struct mft1_note __solo5_mft1_note;
+
 void platform_init(void *arg)
 {
     process_bootinfo(arg);
     fpu_init();
+
+    /*
+     * Get the built-in manifest "out of" the ELF NOTE and validate it. Note
+     * that the size must be adjusted from n_descsz to remove any internal
+     * alignment. Once validated, it is available for access globally by the
+     * bindings.
+     */
+    struct mft *mft = &__solo5_mft1_note.m;
+    size_t mft_size = __solo5_mft1_note.h.n_descsz -
+        (offsetof(struct mft1_note, m) - sizeof (struct mft1_nhdr));
```

Comment:
@mato do all backends have to have code like this, or is there a macro or something that could be used instead?

Reply:
@cfcs Only those that don't have tenders, the others get the mft via different means (`xxx_boot_info`). I'll see what I can do about adding a suitable macro to `mft_abi.h`.